connectors/src/connectors/microsoft/lib/graph_api.ts

```diff
@@ -0,0 +1,84 @@
+import type { Client } from "@microsoft/microsoft-graph-client";
+import type * as MicrosoftGraph from "@microsoft/microsoft-graph-types";
+
+export async function getSites(client: Client) {
+  const res = await client
+    .api("/sites?search=*")
+    .select("id,name,displayName")
+    .get();
+  return res.value as MicrosoftGraph.Site[];
```

Comment:
yes, fun .. their get() return any. We can also just type our methods instead of using `as` , that would work the same way : 

`export async function getSites(client: Client): Promise<MicrosoftGraph.Site[]>`
 
?